Design/Algorithm_ease_of_use.md

```diff
@@ -0,0 +1,88 @@
+Motivation
+==========
+
+Currently code (especially workflow algorithms) will be full of things
+like
+
+```C
+double tolerance = getProperty("CompressTolerance");
+m_outputEW = boost::dynamic_pointer_cast<EventWorkspace>(m_outputW);
+if ((m_outputEW) && (tolerance > 0.)) {
+  g_log.information() << "running CompressEvents(Tolerance=" << tolerance
+                      << ")\n";
+  API::IAlgorithm_sptr compressAlg = createChildAlgorithm("CompressEvents");
+  compressAlg->setProperty("InputWorkspace", m_outputEW);
```

Comment:
This was cut-and-pasted from [AlignAndFocusPowder](https://github.com/mantidproject/mantid/blob/266f2cd000593dd78f061bf6c3c2227857ff169f/Framework/WorkflowAlgorithms/src/AlignAndFocusPowder.cpp#L364). I wanted to make an example from something found in the wild.